### ISSUE TITLE
PowerFSM management in meshhttp

### ISSUE THREAD

**Issue by crossan007** [meshtastic/Meshtastic-device#554]
When a WebUI is connected to the radio, it may or may not be desirable for the radio to enter low-power mode.

This choice should be up to the user.

I don't see any references to `powerFSM` in `meshhttp.cpp`, and have noticed that the radio goes to low power mode when a WebUI is connected (unless the radio is plugged into USB power).

I would like to present through the WebUI an option to prevent the radio from sleeping (even if on battery), but don't see a mechanism to do so.

Proposed solution:
Implement an activity timer within (meshhttp?) which counts seconds since last HTTP request;  if the count exceeds a user-configurable threshold (maybe exposed as a UserSetting setting via protobuf?), then the device is allowed to sleep;  otherwise, it should stay "on" and continue to service HTTP requests

**Comment by mc-hamster** [comment]
Do you have access to the wait_bluetooth_secs preference? That's the setting for the timer that counts down and turns off the radio. If you turn up that value (the default is 60 seconds) then you can set the radio not to turn off.

**Comment by crossan007** [comment]
are HTTP interactions counted towards this timer?

**Comment by mc-hamster** [comment]
they are not, but that's a good point. they should be.

**Comment by Noki** [comment]
Probably an always on mode (disabling all the low power optimizations) for devices directly connected to the powergrid, would be a good idea, especially with plans to extend the mesh to other nodes via the internet.

**Comment by mc-hamster** [comment]
> Probably an always on mode (disabling all the low power optimizations) for devices directly connected to the powergrid, would be a good idea, especially with plans to extend the mesh to other nodes via the internet.

Existing logic is to have the device be "always on" if powered from USB.

**Comment by mc-hamster** [comment]
> are HTTP interactions counted towards this timer?

@crossan007 While you're developing the web interface, please use the change I just checked into my fork. It has the changes for this feature request. You're in a better position to give feedback on the device behavior with this change.

**Comment by Noki** [comment]
@mc-hamster I came to my conclusion because I observed the behavior that the webserver went away without a http connection (#554 not applied) and does not come up again. The serial log shows the following:

![image](https://user-images.githubusercontent.com/198902/102981202-ee992180-4508-11eb-918b-3472e6d5b643.png)

This might be totally unrelated. But I would expect the webserver to be always running on usbPower and bringing it back up from sleep should not fail.

Bringing it back up after a hard-reset works just fine:

![image](https://user-images.githubusercontent.com/198902/102982164-63209000-450a-11eb-9bb1-3164a1c2a24d.png)
